lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h

```diff
@@ -56,7 +57,7 @@ namespace SPIRV {
 template <> inline void SPIRVMap<LinkageType, std::string>::init() {
   add(LinkageTypeExport, "Export");
   add(LinkageTypeImport, "Import");
-  add(LinkageTypeInternal, "Internal");
+  add(LinkageType(internal::LinkageTypeInternal), "Internal");
```

Comment:
To avoid bunch of casts like this you could define `constexpr` variables within `spirv_internal.hpp`, something like:

```c++
enum LinkageTypeInternal {
  LTPrev = LinkageTypeMax - 2,
  LinkageTypeInternal
};

constexpr LinkageType LinkageTypeInternal = static_cast<LinkageType>(LinkagetTypeInternal);
// I hope that static_cast works within constexpr :)
```

BTW, having `LinkageTypeInternal` as both enum name and enum value doesn't seem to be right: it probably works, but if I understand correctly, since this is an unscoped enum, `LinkageTypeInternal` (as value) is also available as global constant, which kind of creates ambiguity with `LinkageTypeInternal` as typename of enum

Reply:
Thanks! Done

Reply:
I think, I'll merged 3 commits when PR is approved.